### PR TITLE
[CHORE] Fixes several errors introduced by #2110

### LINKED ISSUE #2271: [CHORE] Clean-up string parsing from templates to Typescript
Currently we perform quite some string parsing from the HTML templates to the Typescript code. For example, to return the user a message that they haven't enter a name we get this string in the template though Jinja and then add it as an argument to a Typescript function. 

This is cumbersome and likely to introduce bugs (such as quotes in the strings). We can escape all these errors by surrounding them with `|tojson` parsing but this does not make any sense. We should dependent on the server to return specific messages, not on our own "houtje-touwtje" string parsing. This is introduced due to merging #2210 and losing access to all translation in the Typescript and should be cleaned-up now.

First things that come to mind that need clean-up:

- Creation of a class
- Renaming of a class
- Inviting a student
- Change password student
- Creating an adventure
- Basically anything that uses the `modal.prompt` and verifies for an empty string

When we want to keep the string we should always enclose it with `|tojson` to prevent quotes and other special characters from escaping the function.

### PR DESCRIPTION
**Pull request by TiBiBa**
**Description**
In this PR we fix several errors introduced by the merging of #2210. As some of them are functionality breaking, we should merge this before deploying main to the production version. A small summation of fixes:

- Several still existing `g.ui_texts.get` calls are removed (fixed pages crashing)
- Classes can be created and renamed again when on English
    - This was not possible due to the string parsed to the front-end containing a quote (and closing the function)
    - We parse all relevant jinja variables with `|tojson` to prevent this from happening again
- We move the empty name validation for the class, adventure and student invite to the back-end
    - This way reducing the string "ping pong" on the front-end and improve security overall
- Added some missing gettext() calls that where still non-translated strings
- Removed some redundant to-do's as they are fixed by #2110

**Fixes**
This PR fixes #2271

**How to test**
This is a clean-up task. However, it can be verified that the following functionality should work again in English (as it currently does not working on Alpha):

- Creating and renaming a class
- Creating and renaming an adventure
- Inviting a student

### CODE REVIEW

**Review by boryanagoncharenko — COMMENTED** [review]
Approving since the u-string discussion is broader than the context of this PR.

File: website/teacher.py [hunk 1]
```diff
@@ -160,7 +164,7 @@ def prejoin_class (class_id, link):
                 user = DATABASE.user_by_username(token ['username'])
 
         return render_template ('class-prejoin.html', joined=False,
-                                page_title=hedyweb.g.ui_texts.get('title_join-class'), current_page='my-profile',
+                                page_title=gettext(u'title_join-class'), current_page='my-profile',
```
[the remark this answers is not in the document]
Comment (Felienne) [reply website/teacher.py:167]: Yeah I think you are right @boryanagoncharenko, the u is not needed I think, unless Babel has something special going on.

**Review by Felienne — APPROVED** [review]

**Comment by TiBiBa** [comment]
Will fix the merge conflicts and then I think we should merge this with some priority as some functionality is currently broken.

**Comment by boryanagoncharenko** [comment]
I thought I approved the PR but it seems that I only left a comment and enabled auto-merging. Apologies about that.

**Pull request merged**